lib/grandpa/network.go

```diff
@@ -225,41 +225,20 @@ func (s *Service) sendNeighbourMessage() {
 
 // decodeMessage decodes a network-level consensus message into a GRANDPA VoteMessage or CommitMessage
 func decodeMessage(msg *ConsensusMessage) (m GrandpaMessage, err error) {
-	var (
-		mi interface{}
-		ok bool
-	)
-
-	switch msg.Data[0] {
-	case voteType:
-		r := &bytes.Buffer{}
-		_, _ = r.Write(msg.Data[1:])
-		vm := &VoteMessage{}
-		err = vm.Decode(r)
-		m = vm
+	err = scale.Unmarshal(msg.Data, &gossipMessage)
```

Comment:
we need to check this error, right?

Reply:
I have it below the switch, but I can move it up if that's better

Reply:
would be nice to check the error before the switch, will be easier to relate the error to the `Unmarshal`

Reply:
Yes please handle the error immediately